src/mail.sh

```diff
@@ -63,10 +67,42 @@ function validate_encryption()
   fi
 }
 
+# This function checks that the necessary options to use
+# the git send-email have been set
+function mail_verify()
+{
+  local count=0
+  local -a missing_conf
+  local set_confs
+
+  set_confs=$(git config --get-regexp 'user|sendemail\.smtp')
+  for config in "${essential_config_options[@]}"; do
+    if ! printf '%s\n' "$set_confs" | grep -cF "$config" - &> /dev/null; then
+      missing_conf[$count]="$config"
+      count=$((count + 1))
+    fi
+  done
+
+  for config in "${recommended_config_options[@]}"; do
+    if ! printf '%s\n' "$set_confs" | grep -cF "$config" - &> /dev/null; then
+      missing_conf[$count]="$config"
+      count=$((count + 1))
+    fi
+  done
+
+  if [[ $count -gt 0 ]]; then
+    complain "Missing configurations needed for send-email: ${missing_conf[*]}"
+    return 1
```

Comment:
I think we have two cases:
1. Essential: We must fail if something is missing.
2. Recommended: It is not ideal, but we can proceed since it will not compromise the` mail` feature. Let's just warn the user.